### PR TITLE
impl(otel): no spans if no sleeps

### PR DESCRIPTION
**Pull request by dbolduc**
Motivated by #12959 

I still like invoking the `sleeper` even when the duration is 0.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13125)

### CODE REVIEW

**Comment by devbww** [comment]
> I still like invoking the `sleeper` even when the duration is 0.

I agree.  Even if there is no sleeping, it may have other side effects (despite this "sleeper" deciding that it won't).

**Review by devbww — APPROVED** [review]

File: google/cloud/internal/rest_retry_loop_test.cc [hunk 1]
```diff
@@ -58,8 +58,8 @@ std::unique_ptr<RetryPolicy> TestRetryPolicy() {
 }
 
 std::unique_ptr<BackoffPolicy> TestBackoffPolicy() {
-  return ExponentialBackoffPolicy(std::chrono::microseconds(1),
```
Comment (coryan) [inline google/cloud/internal/rest_retry_loop_test.cc:61]: Are these changes related at all?

Comment (dbolduc) [reply google/cloud/internal/rest_retry_loop_test.cc:61]: Yeah. The backoff policy returns something in milliseconds, which rounds to 0.

https://github.com/googleapis/google-cloud-cpp/blob/fce4d6864130ec4c0396190ad13ca5537e6f2e1a/google/cloud/internal/backoff_policy.h#L72
